trpc/router/goal.ts

```diff
@@ -358,8 +358,44 @@ export const goal = router({
             throw new TRPCError({ code: 'INTERNAL_SERVER_ERROR', message: String(error.message), cause: error });
         }
     }),
-    changeProject: protectedProcedure.input(goalChangeProjectSchema).mutation(async ({ input }) => {
-        return changeGoalProject(input.id, input.projectId);
+    changeProject: protectedProcedure.input(goalChangeProjectSchema).mutation(async ({ ctx, input }) => {
+        const actualGoal = await prisma.goal.findUnique({
+            where: { id: input.id },
+        });
+
+        if (!actualGoal) return null;
+
+        const { activityId } = ctx.session.user;
+
+        try {
+            await changeGoalProject(input.id, input.projectId);
+            const goal = await prisma.goal.update({
+                where: {
+                    id: input.id,
+                },
+                data: {
+                    history: {
+                        create: {
+                            activityId,
+                            subject: 'project',
+                            action: 'change',
+                            previousValue: actualGoal.projectId,
+                            nextValue: input.projectId,
+                        },
+                    },
+                },
+                include: {
+                    ...goalDeepQuery,
+                },
+            });
+
+            return {
+                ...goal,
+                ...addCalclulatedGoalsFields(goal, activityId),
+            };
+        } catch (error: any) {
```

Comment:
I think I've seen this catch already, maybe it should be a helper function?

Reply:
https://github.com/taskany-inc/issues/issues/1216